sigauth/signature.go

```diff
@@ -116,7 +116,11 @@ func (s *SignatureAuth) SignRequest(r *http.Request) error {
 
 	headers := []string{"request-line", "host", "date", "digest", "content-length"}
 
-	sigAuthHeader := SignatureAuthHeader{KeyID: s.serviceRegManager.AuthService.ServiceID, Algorithm: "rsa-sha256", Headers: headers}
+	serviceReg, err := s.serviceRegManager.GetServiceRegWithPubKey(s.serviceRegManager.AuthService.ServiceID)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve service registration: %v", err)
+	}
```

Comment:
There is no guarantee that the signature being generated is using a key associated with a service reg record. Here we need to use the fingerprint of the `s.serviceKey` directly rather than using service reg records.